src/lib/reasoners/shostak.ml

```diff
@@ -500,9 +500,11 @@ struct
       List.filter (fun (p,_) ->
           match p.v with
           | Ac _ -> true | Term _ -> SX.mem p original
-          | _ ->
-            Printer.print_err "%a" CX.print p;
-            assert false
```

Comment:
@iguerNL 
I did not find what enforces this invariant ; do you remember what should have forced it to be either Ac or a Term ?